Stripe/STPAddCardViewController.m

```diff
@@ -279,10 +284,16 @@ - (void)nextPressed:(__unused id)sender {
     }
     // Create and return a Payment Method
     STPPaymentMethodBillingDetails *billingDetails = [[STPPaymentMethodBillingDetails alloc] init];
-    billingDetails.address = [[STPPaymentMethodAddress alloc] initWithAddress:self.addressViewModel.address];
-    billingDetails.email = self.addressViewModel.address.email;
-    billingDetails.name = self.addressViewModel.address.name;
-    billingDetails.phone = self.addressViewModel.address.phone;
+    if (self.configuration.requiredBillingAddressFields == STPBillingAddressFieldsPostalCode) {
+        STPAddress *address = [[STPAddress alloc] init];
+        address.postalCode = self.paymentCell.paymentField.postalCode;
+        billingDetails.address = [[STPPaymentMethodAddress alloc] initWithAddress:address];
+    } else {
+        billingDetails.address = [[STPPaymentMethodAddress alloc] initWithAddress:self.addressViewModel.address];
```

Comment:
does this include the postal code?

Reply:
It does: The STPAddressViewModel handles the ZIP when using the full address entry form, but the STPPaymentCardTextFieldViewModel handles it when we're only requesting the ZIP. It's not the cleanest implementation, but the UX is better than breaking out ZIP into a separate field below the CC entry field.